compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -3955,6 +3956,15 @@ declare_lint! {
     "`break` expression with label and unlabeled loop as value expression"
 }
 
+crate::declare_tool_lint! {
+    /// The `broken_mir` statically detects undefined behaviour in the MIR optimization pipeline.
+    /// This is an internal lint, and not intended to be used directly.
+    pub rustc::BROKEN_MIR,
+    Deny,
```

Comment:
Forbid maybe?

Reply:
One use case is to use `expect`. And you cannot expect a forbidden lint.